packages/addon-dev/src/rollup-public-entrypoints.ts

```diff
@@ -15,15 +19,37 @@ export default function publicEntrypoints(args: {
     name: 'addon-modules',
     async buildStart() {
       let matches = walkSync(args.srcDir, {
-        globs: [...args.include],
+        globs: [...args.include, hbsPattern],
       });
 
       for (let name of matches) {
-        this.emitFile({
-          type: 'chunk',
-          id: join(args.srcDir, name),
-          fileName: normalizeFileExt(name),
-        });
+        if (args.include.some((pattern) => minimatch(name, pattern))) {
+          // anything that matches one of the user's patterns is definitely emitted
+          this.emitFile({
+            type: 'chunk',
+            id: join(args.srcDir, name),
+            fileName: normalizeFileExt(name),
+          });
+        } else {
+          // this file didn't match one of the user's patterns, so it must match
+          // our hbsPattern. Infer the possible existence of a synthesized
+          // template-only component JS file and test whether that file would
+          // match the user's patterns.
+          let normalizedName = normalizeFileExt(name);
+          let id = join(args.srcDir, normalizedName);
+          if (
+            args.include.some((pattern) =>
+              minimatch(normalizedName, pattern)
+            ) &&
+            !pathExistsSync(id)
+          ) {
+            this.emitFile({
```

Comment:
Ah,. This was probably what i was missing (and maybe all the changes in this file).


Because i was so focused on the other plugin, i didn't remember how in interacts with this plugin.